SConstruct

```diff
@@ -179,18 +177,43 @@ for k in platform_opts.keys():
     for o in opt_list:
         opts.Add(o)
 
-for x in module_list:
-    module_enabled = True
-    tmppath = "./modules/" + x
-    sys.path.insert(0, tmppath)
+# Detect modules.
+opts.Add("custom_modules", "Directory path containing custom modules to build along with built-in modules.", "")
```

Comment:
Should mention that it can be a list of paths (comma-separated if we go with that, or `os.pathsep` separated otherwise, but as mentioned this makes documentation and usage trickier if users need to call `python -c "import os; print(os.pathsep)"` from their shell to have a cross-platform call).